### ISSUE TITLE
Parsing DynamoDB attribute values from KinesisClientRecord using KCL V2

### ISSUE THREAD

**Issue by nihilson** [awslabs/amazon-kinesis-client#856]
I am following this KCL v2 java [sample](https://docs.aws.amazon.com/streams/latest/dev/kcl2-standard-consumer-java-example.html) to create a Consumer application for reading the change data capture records of DynamoDB. It works all good. I am able to read the data from the KinesisClientRecord object and print the String as below : 

`{"awsRegion":"us-east-1","eventID":"799c8613-1a53-4692-8411-222c5062d6c0","eventName":"INSERT","userIdentity":null,"recordFormat":"application/json","tableName":"TableName","dynamodb":{"ApproximateCreationDateTime":1633093731707,"Keys":{"PK":{"S":"some-value"},"SK":{"S":"some-value"}},"NewImage":{"SK":{"S":"some-value"},"EXPIRE":{"N":"1664629731"},"CREATED":{"N":"1633093731"},"PK":{"S":"some-value"},"DATA":{"S":"some-value"}},"SizeBytes":164},"eventSource":"aws:dynamodb"}`

Came across this [sample](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/Streams.KCLAdapter.Walkthrough.html#Streams.KCLAdapter.Walkthrough.Step3) that describes how to use the DynamoDB Streams Adapter to read the DynamoDB item attribute values from the records using "streamRecord.getDynamodb().getNewImage() " construct.

But I am unable to use the DynamoDB Streams Adapter as the KinesisClientRecord is not compatible with DDB Record. I assume KCL V2 does not support it. Looking for some cleaner option to parse and extract the DDB attribute values. Please provide any suggestions or code samples to read the attribute values. 

```
public void processRecords(ProcessRecordsInput processRecordsInput) {
    MDC.put(SHARD_ID_MDC_KEY, shardId);
    try {
        log.info("Processing {} record(s)", processRecordsInput.records().size());
        for( KinesisClientRecord record : processRecordsInput.records()) {
            String recordData = null;
            try {
                recordData = decoder.decode(record.data()).toString();
                System.out.println( recordData );
                // TODO : Looking for an adapter class to convert the KinesisClientRecord compatible to DynamoDB ResourceAdapater to read the attribute values like PK, SK, DATA in a clean way. 
            } catch (CharacterCodingException e1) {
                // TODO Auto-generated catch block
                e1.printStackTrace();
            }                
        }
    } catch (Throwable t) {
        log.error("Caught throwable while processing records. Aborting.");
        Runtime.getRuntime().halt(1);
    } finally {
        MDC.remove(SHARD_ID_MDC_KEY);
    }
}
```

**Comment by j-ohara** [comment]
I am also facing this issue - it would be very nice to do this cleanly with KCL V2.

**Comment by andklee** [comment]
```
software.amazon.awssdk.services.dynamodb.model.Record ddbRecord = objMapper.readValue(record.getData().array(),software.amazon.awssdk.services.dynamodb.model.Record.serializableBuilderClass()).build();
software.amazon.awssdk.services.dynamodb.model.StreamRecord streamRecord = ddbRecord.dynamodb();
```

**Comment by kurczynski** [comment]
Yeah, this is definitely more than just a "nice to have" because it throws more repetitive work onto devs.

> You can use jackson mapper to map the record to its class. Then you can access the methods directly.
> 
> ```
> 
> private static ObjectMapper objMapper = new ObjectMapper()
>           .configure(MapperFeature.ACCEPT_CASE_INSENSITIVE_PROPERTIES, true)
>           .configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false)
>           .findAndRegisterModules();
> ...
> software.amazon.awssdk.services.dynamodb.model.Record ddbRecord = objMapper.readValue(record.getData().array(),software.amazon.awssdk.services.dynamodb.model.Record.serializableBuilderClass()).build();
> software.amazon.awssdk.services.dynamodb.model.StreamRecord streamRecord = ddbRecord.dynamodb();
> 
> ByteBuffer newBytesBuffer = streamRecord.newImage().get("binaryData").b().asByteBuffer();
> ```

This won't work because `record.data()` (the method `record.getData()` doesn't exist) refers to a read-only `ByteBuffer` so it'll throw a `java.nio.ReadOnlyBufferException`. You _can_ serialize it to JSON and then deserialize it to a `Record` object:
```
CharsetDecoder charsetDecoder = StandardCharsets.UTF_8.newDecoder();
String json = charsetDecoder.decode(record.data()).toString();
Record recordObject = objectMapper.readValue(json, Record.serializableBuilderClass()).build();
```

But again, that's more repetitive work that the API puts on devs.
